src/main/java/reposense/model/RepoLocation.java

```diff
@@ -16,14 +14,21 @@
  * Represents a repository location.
  */
 public class RepoLocation {
-    private static final String GIT_LINK_SUFFIX = ".git";
     private static final String MESSAGE_INVALID_LOCATION = "%s is an invalid location.";
+    private static final String MESSAGE_INVALID_REMOTE_URL = "%s is an invalid remote URL.";
+
     private static final Pattern GIT_REPOSITORY_LOCATION_PATTERN =
-            Pattern.compile("^https?://github.com/(?<org>.+?)/(?<repoName>.+?)\\.git$");
+            Pattern.compile("^(ssh|git|https?|ftps?)://[^/]*?/(?<path>.*?)/?(?<repoName>[^/]+?)(/?\\.git)?/?$");
```

Comment:
Are we also able to add support for `file://` protocol? Git treats it as a local path when cloning.

Reply:
Hi thanks for reminding me about this! I initially supported it then removed it because it caused quite a bit of issue being a URL and being local. However, I think the current implementation should support it easily. I will update it when I get back. Thanks!

Reply:
@dcshzj I've updated the implementation to support `file://` as well now. Would like to request a follow-up review. Thanks!